### DISCUSSION TITLE
Preserving the history of Input Leap and its Synergy roots

### DISCUSSION THREAD

**Discussion by ccoenen** [input-leap/input-leap#1633] [General]
With the origins of input leap being more than two decades old, it has become quite hard to learn the whole story of this project. I started collecting its predecessors here: https://github.com/debauchee/barrier/issues/109#issuecomment-1496057294 but I think this story deserves a more permanent place, and perhaps even some deeper digging into who contributed what to the project.

As far as I can tell (and as far as I was able to rememer), this is the chain of events that lead to input-leap:
* apparently, in 1996 this was created as CosmoSynergy at SGI. (taken from wikipedia article, I have not used this myself)
* early 2000s until ~2006 (release 1.3.1): Synergy by [Chris Schoenemann and a few contributors](https://web.archive.org/web/20070630222420if_/http://synergy2.sourceforge.net/authors.html) - hosted on sourceforge in CVS. This is the first version I actually used.
* 2009 until 2010: Synergy+ by [Sorin Sbârnea and later Nick Bolton and a few more](https://web.archive.org/web/20090319014135/http://code.google.com/p/synergy-plus/) -hosted on Google Code
* Synergy+ and Synergy joining into Synergy-Foss (late 2010 [on redmine](https://web.archive.org/web/20101014181637/http://synergy-foss.org/)) and Synergy-Project (2014, [own website](https://web.archive.org/web/20141109020711/http://synergy-project.org/) and [code on github under the name synergy, not symless](https://web.archive.org/web/20141114011053/https://github.com/synergy)), which later became Symless with Synergy as their product.
* 2016 this became [Symless](https://web.archive.org/web/20160630194628/http://symless.com/synergy/) with Synergy as a product with binaries being sold.
* In 2018, Barrier made [the first releases](https://github.com/debauchee/barrier/releases/tag/v2.0.0-RC1) as a fork from symless/synergy
* In 2022, Input-Leap was forked from Barrier

**Comment by ccoenen** [comment]
Coincidentally, we're just barely missed the 20th birthday of the 1.0.0 release of Synergy: In [January 2003, the latest release was 0.9.13](https://web.archive.org/web/20030115213012/http://sourceforge.net:80/news/?group_id=59275), and the [next snapshpot from February 2003 already lists 1.0.2](https://web.archive.org/web/20030210160601/http://sourceforge.net:80/news/?group_id=59275)

**Comment by shymega** [comment]
@ccoenen Thank you for this post. I would really like to see this in the wiki, where it can be expanded upon in its own, special page about the history of this piece of software.

I would like to suggest that you, or someone else interested, make a PR to this repo - https://github.com/input-leap/wiki-prs - on a new branch, where it'll be merged into the wiki. This is how we make changes to the wiki, as making it RW for everyone would be disastrous.

**Reply by ccoenen** [reply]
finally got around to actually making the edits: https://github.com/input-leap/wiki-prs/pull/4

**Comment by ccoenen** [comment]
I would also like to invite @crs23 in here, maybe he has some more insights into the early development of Synergy?
